src/Traversals/depthfirst.jl

```diff
@@ -142,3 +142,52 @@ end
 end
 
 @traitfn is_cyclic(g::::(!IsDirected), alg::DepthFirst=DepthFirst()) = ne(g) > 0
+
+############ toposort2
+#
+mutable struct TopoSortState2{T<:Integer} <: AbstractTraversalState
+    vcolor::Vector{UInt8}
+    verts::Vector{T}
+    targetvert::T
+    nneighs::T
+end
+
+function previsitfn!(s::TopoSortState2, u)
+    s.targetvert = u
+    s.nneighs = 0
+    println("targeting $u")
+    return true
+end
+
+function newvisitfn!(s::TopoSortState2, u, v)
+    s.nneighs += 1
+    println("nneighs now $(s.nneighs)")
+    s.vcolor[v] = max(1, s.vcolor[v])
+    return true
+end
+
+function postlevelfn!(s::TopoSortState2)
+    if s.nneighs == 0 && s.vcolor[s.targetvert] != 2
+        println("no neighbors")
+        push!(s.verts, s.targetvert)
+        s.vcolor[s.targetvert] = 2
+        s.targetvert = 0
+        s.nneighs = 0
```

Comment:
This line is redudant

Reply:
Come to think of it, is `s.targetvert  = 0` not also redudant? It looks as it will be reset by `previsitfn!` anyway.